Framework/MDAlgorithms/src/GetSpiceDataRawCountsFromMD.cpp

```diff
@@ -499,17 +501,12 @@ MatrixWorkspace_sptr GetSpiceDataRawCountsFromMD::createOutputWorkspace(
     throw std::runtime_error("Failed to create output matrix workspace.");
 
   // Set data
-  MantidVec &dataX = outws->dataX(0);
-  MantidVec &dataY = outws->dataY(0);
-  MantidVec &dataE = outws->dataE(0);
-  for (size_t i = 0; i < sizex; ++i) {
-    dataX[i] = vecX[i];
-    dataY[i] = vecY[i];
-    if (dataY[i] > 1.)
-      dataE[i] = sqrt(dataY[i]);
-    else
-      dataE[i] = 1.;
-  }
+  outws->setPoints(0, vecX);
+  outws->setCounts(0, vecY);
```

Comment:
Agreed about the setting method. But here this is returning `1.0` if the Y value is `< 1.0` are well as performing the square root? So calculating it automatically could change the algorithm's output?